confluence-xml/src/main/java/org/xwiki/contrib/confluence/filter/internal/input/ConfluenceConverterListener.java

```diff
@@ -66,18 +66,18 @@
 @InstantiationStrategy(ComponentInstantiationStrategy.PER_LOOKUP)
 public class ConfluenceConverterListener extends WrappingListener
 {
-    private static final Pattern PATTERN_URL_DISPLAY = Pattern.compile("^/display/(.+)/([^\\?#]+)(\\?.*)?$");
+    private static final Pattern PATTERN_URL_DISPLAY = Pattern.compile("^/display/(.+)/([^?#]+)(\\?.*)?$");
```

Comment:
I'm surprised `?` does not need to be escaped. If test are passing, I guess all is well.